### PR TITLE
adding stride_order before convolution

### PR DESCRIPTION
**Pull request by jjsjann123**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: thunder/executors/torchex.py [hunk 1]
```diff
@@ -1235,6 +1235,10 @@ def _convolution_transform(
     output_padding: Sequence[int],
     groups: int,
 ) -> TensorProxy:
+    if dtypes.is_low_precision_dtype(a.dtype):
+        if a.ndim == 4 or a.ndim == 5:
+            a = prims.stride_order(a, (a.ndim - 1, 0) + tuple(range(a.ndim - 2, 0, -1)))
+            weight = prims.stride_order(weight, (a.ndim - 1, 0) + tuple(range(a.ndim - 2, 0, -1)))
     return convolution(a, weight, bias, stride, padding, dilation, bool(transposed), output_padding, groups)
```
Comment (nikitaved) [inline thunder/executors/torchex.py:1242]: Shouldn't the result of `convolution` be restrided as well?

**Comment by jjsjann123** [comment]
numerical issue looks a bit annoying. Changing the torch reference implementation using channels last. I thought this is automatically done in cudnn (NOT VERIFIED!), but there's still places where torch convolution goes through native kernels.

**Comment by t-vi** [comment]
@jjsjann123 what's your view on this? The CI seemed to pass earlier, but the PR is still a draft and thus not merged...

**Comment by jjsjann123** [comment]
sorry about keeping this one floating. I haven't got any chance to revist its impact on vision models, so I wasn't confidently pushing for it.

I'll close it for now and revisit when I have a bit more bandwidth.

**Pull request closed without merging**